### PR TITLE
FIX: decouple msk auth from glue auth in KafkaSource

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR decouples MSK aws auth from that for AWS Glue
 
### Issues Resolved
Resolves #[Issue number to be closed when this PR is merged]
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSecurityConfigurerTest.java [hunk 1]
```diff
@@ -128,6 +142,133 @@ public void testSetAuthPropertiesAuthSslWithNoCertContentNoTrustStore() throws E
...
+                is("software.amazon.msk.auth.iam.IAMClientCallbackHandler"));
+    }
+
+    @Test
+    public void testSetAuthPropertiesBootstrapServersWithSaslIAMDefault() throws IOException {
+        final Properties props = new Properties();
+        final KafkaSourceConfig kafkaSourceConfig = createKafkaSinkConfig("kafka-pipeline-bootstrap-servers-sasl-iam-default.yaml");
+        KafkaSecurityConfigurer.setAuthProperties(props, kafkaSourceConfig, LOG);
+        assertThat(props.getProperty("bootstrap.servers"), is("localhost:9092"));
+        assertThat(props.getProperty("sasl.mechanism"), is("AWS_MSK_IAM"));
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSecurityConfigurerTest.java:169]: Should there be an assertion for the `sasl.jaas.config` here as well?

File: data-prepper-plugins/kafka-plugins/src/test/resources/kafka-pipeline-bootstrap-servers-sasl-iam-role.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        bootstrap_servers:
+          - "localhost:9092"
+        encryption:
+          type: "SSL"
+        authentication:
+          sasl:
+            aws_msk_iam: role
+        aws:
+          region: us-east-2
+          sts_role_arn: sts_role_arn
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/test/resources/kafka-pipeline-bootstrap-servers-sasl-iam-role.yaml:13]: Can you use something more unique to help ensure the assertion in the test is unique. Maybe `test_sasl_iam_sts_role`?

File: data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSecurityConfigurerTest.java [hunk 3]
```diff
@@ -128,6 +142,133 @@ public void testSetAuthPropertiesAuthSslWithNoCertContentNoTrustStore() throws E
...
+        assertThat(props.getProperty("ssl.truststore.location"), is(nullValue()));
+        assertThat(props.getProperty("ssl.truststore.password"), is(nullValue()));
+        assertThat(props.get("ssl.engine.factory.class"), is(nullValue()));
+        assertThat(props.get("sasl.client.callback.handler.class"),
+                is("software.amazon.msk.auth.iam.IAMClientCallbackHandler"));
+    }
+
+    @Test
+    public void testSetAuthPropertiesBootstrapServersOverrideByMSK() throws IOException {
+        final String testMSKEndpoint = "bootstrap-server-string";
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSecurityConfigurerTest.java:181]: Maybe this a random UUID.

File: data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSecurityConfigurerTest.java [hunk 4]
```diff
@@ -128,6 +142,133 @@ public void testSetAuthPropertiesAuthSslWithNoCertContentNoTrustStore() throws E
...
+        assertThat(props.getProperty("ssl.truststore.location"), is(nullValue()));
+        assertThat(props.getProperty("ssl.truststore.password"), is(nullValue()));
+        assertThat(props.get("ssl.engine.factory.class"), is(nullValue()));
+        assertThat(props.get("sasl.client.callback.handler.class"),
+                is("software.amazon.msk.auth.iam.IAMClientCallbackHandler"));
+    }
+
+    @Test
+    public void testSetAuthPropertiesMskWithSaslPlain() throws IOException {
+        final String testMSKEndpoint = "bootstrap-server-string";
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/test/java/org/opensearch/dataprepper/plugins/kafka/util/KafkaSecurityConfigurerTest.java:211]: Also, make this a random UUID string.

File: data-prepper-plugins/kafka-plugins/src/test/resources/kafka-pipeline-msk-sasl-plain.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+log-pipeline :
+  source:
+     kafka:
+        encryption:
+          type: "SSL"
+        authentication:
+          sasl:
+            plain:
+              username: username
```
Comment (dlvenable) [inline data-prepper-plugins/kafka-plugins/src/test/resources/kafka-pipeline-msk-sasl-plain.yaml:9]: Use some values that are more unique: `test_sasl_username` and `test_sasl_password` would work.

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thanks @chenqi0805 for this change! Overall it looks good. I have a few comments first.

**Review by dlvenable — APPROVED** [review]

**Review by kkondaka — APPROVED** [review]

**Pull request merged**